File: java/problem1.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> h = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int n = target - nums[i];
            if (h.containsKey(n)) {
                return new int[] { h.get(n), i };
            } else {
                h.put(nums[i], i);
            }
        }
        throw new IllegalArgumentException("No two sum solution");
    }
}

public class problem1 {
    public static void main(String[] args) {
        Solution x = new Solution();
        int[] a = { -3, 4, 3, 90 };
        for (int i : x.twoSum(a, 0)) {
            System.out.print(i + " ");
        }
    }
}

File: cpp/problem1.cpp
#include <iostream>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
   public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> h;
        for (int i = 0; i < nums.size(); ++i) {
            int n = target - nums[i];
            if (h.find(n) != h.end()) {
                return {h[n], i};
            } else {
                h[nums[i]] = i;
            }
        }
        throw "No two sum solution";
    }
};

int main() {
    vector<int> a = {2, 7, 11, 15};
    vector<int> ivec = Solution().twoSum(a, 9);
    for (int c : ivec) cout << c << " ";
    cout << "\n";
    return 0;
}
